include/boost/geometry/algorithms/detail/overlay/colocate_clusters.hpp

```diff
@@ -0,0 +1,108 @@
+// Boost.Geometry
+
+// Copyright (c) 2023 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_OVERLAY_COLOCATE_CLUSTERS_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_OVERLAY_COLOCATE_CLUSTERS_HPP
+
+#include <type_traits>
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/coordinate_dimension.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace overlay
+{
+
+// Default implementation, using the first point for all turns in the cluster.
+template
+<  
+    typename Point,
+    typename CoordinateType = typename geometry::coordinate_type<Point>::type,
+    typename CsTag = typename geometry::cs_tag<Point>::type,
+    bool IsIntegral = std::is_integral<CoordinateType>::value
+>            
+struct cluster_colocator 
+{
+    template <typename TurnIndices, typename Turns>
+    static inline void apply(TurnIndices const& indices, Turns& turns)
+    {
+        // This approach works for all but one testcase (rt_p13)
+        // The problem is fill_sbs, which uses sides and these sides might change slightly
+        // depending on the exact location of the cluster.
+        // Using the centroid is, on the average, a safer choice for sides.
+        // Alternatively fill_sbs could be revised, but that requires a lot of work
+        // and is outside current scope.
+        // Integer coordinates are always colocated already and do not need centroid calculation.
+        // Geographic/spherical coordinates might (in extremely rare cases) cross the date line
+        // and therefore the first point is taken for them as well.
+        auto it = indices.begin();
+        auto const& first_point = turns[*it].point;
+        for (++it; it != indices.end(); ++it)
+        {
+            turns[*it].point = first_point;
+        }
```

Comment:
Because all implementation is here, I removed the define, better like this.
Thanks for the comments.